main_data_processor.py

```diff
@@ -1,2 +1,210 @@
+        description=(
+            "DANE worker that extracts audio files from video files (for further processing in other DANE workers)"
+        ),
+        input_data={"input_file_path": input_file_path},
+        parameters=dict(cfg.AUDIO_EXTRACTION_SETTINGS),
+        software_version=obtain_software_versions(DANE_WORKER_ID),
+    )
+    provenance_chain = []  # will contain the steps of the top-level provenance
+
+    # S3 URI, local tar.gz or locally extracted tar.gz is allowed
+    if validate_s3_uri(input_file_path):
+        model_input = obtain_input_file(input_file_path)
+    else:
+        if input_file_path.find(".tar.gz") != -1:
+            source_id = get_source_id_from_tar(input_file_path)
+        else:
+            source_id = input_file_path.split("/")[-2]
+
+        model_input = AudioExtractionInput(
+            200,
+            f"Processing tar.gz archive: {input_file_path}",
+            source_id,
+            input_file_path,
+            None,  # no download provenance when using local file
+        )
+
+    # add the download provenance
+    if model_input.provenance:
+        provenance_chain.append(model_input.provenance)
+
+    # first generate the output dirs
+    generate_output_dirs(model_input.source_id)
+
+    # apply model to input & extract features
+    proc_result = apply_model(model_input)
+
+    if proc_result.provenance:
+        provenance_chain.append(proc_result.provenance)
+
+    # as a last piece of output, generate the provenance.json before packaging&uploading
+    full_provenance_chain = stop_timer_and_persist_provenance_chain(
+        provenance=top_level_provenance,
+        output_data={
+            "output_path": get_base_output_dir(model_input.source_id),
+            "output_uri": get_s3_output_file_uri(model_input.source_id),
+        },
+        provenance_chain=provenance_chain,
+        provenance_file_path=get_output_file_path(
+            model_input.source_id, OutputType.PROVENANCE
+        ),
+    )
+
+    # if all is ok, apply the I/O steps on the outputted features
+    validated_output: CallbackResponse = apply_desired_io_on_output(
+        model_input,
+        proc_result,
+        cfg.INPUT.DELETE_ON_COMPLETION,
+        cfg.OUTPUT.DELETE_ON_COMPLETION,
+        cfg.OUTPUT.TRANSFER_ON_COMPLETION,
+    )
+    logger.info("Results after applying desired I/O")
+    logger.info(validated_output)
+    return validated_output, full_provenance_chain
+
+
+def apply_model(
+    video_input: AudioExtractionInput,
+) -> AudioExtractionOutput:
+    logger.info("Starting model application")
+    start = time.time() * 1000  # convert to ms
+    destination = get_output_file_path(video_input.source_id, OutputType.AUDIO)
+    ffmpeg_cmd = ["ffmpeg", "-i", video_input.input_file_path]
+    if cfg.AUDIO_EXTRACTION_SETTINGS.CONVERT_TO_MONO:
+        ffmpeg_cmd = ffmpeg_cmd + ["-ac", "1"]
+    if cfg.AUDIO_EXTRACTION_SETTINGS.OUTPUT_SAMPLERATE_HZ != "None":
```

Comment:
Somehow, the following is executed in line 132:
['ffmpeg', '-i', '/data/input-files/testsource__testcarrier/testsource__testcarrier.mp4', '-ar', None, '/data/output-files/testsource__testcarrier/audio/testsource__testcarrier.wav']

So something seems to go wrong in this check

Reply:
You might consider also to use 0 to signal "None" rather than a string, so that this variable will always be numeric

Reply:
good idea, will do that